### PR TITLE
Parse shader file from config and add json library

### PR DESCRIPTION
**Pull request by jonmgomez**
Only really need to look at main.cpp here. This is adding a json parsing library and reading in a config file form command arguments. Use the "shader" key to load the shader into the program.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,13 +31,15 @@ add_executable(pong ${SOURCES})
 add_subdirectory(glfw-3.3.8)
 add_subdirectory(glew-2.1.0)
 add_subdirectory(glm-0.9.9.8)
+add_subdirectory(json-3.11.2/json-3.11.2)
```
Comment (jonmgomez) [inline CMakeLists.txt]: Curious what is the best way to organize tps libraries for this project @cjappl? I know we used zip files in some way before? Looking to include these in a nice way, that doesn't make this project show as being mainly an HTML repo haha

Comment (cjappl) [reply CMakeLists.txt]: So this is a great time to reorganize your tps. I highly recommend you zip all of these dependencies into tar.gz files to save space (and ensure you don't have some crazy 2k line PR). I would recommend you add a `tps` top level folder with it's own CMakeLists.txt

From there, the pattern is like so in your tps/CMakeLists.txt:

```
FetchContent_Declare(doctest
    URL ${CMAKE_CURRENT_SOURCE_DIR}/doctest/doctest-2.4.11.tar.gz
)
FetchContent_MakeAvailable(doctest)

```

What this does is automagically unzip/untar the thing, then `FetchContent_MakeAvailable` goes through and calls `add_subdirectory` like you are doing here

If you ever want to play around with a dependency without adding it to your repo, or you don't want to check something in, FetchContent works on git repos that are built with CMake:

```
# Fetch zlib from its Git repository
FetchContent_Declare(
    zlib
    GIT_REPOSITORY https://github.com/madler/zlib.git
    GIT_TAG v1.2.11  # You can specify a specific tag or commit hash
)
```

File: main.cpp [hunk 2]
```diff
@@ -8,12 +8,38 @@
...
+        return nullptr;
+    }
+
+    nlohmann::json jsonData;
+
+    try
+    {
+        jsonFile >> jsonData;
+    }
+    catch (const nlohmann::json::parse_error& e)
```
Comment (cjappl) [inline main.cpp:34]: EXCEPTIONS??????????????????? 🤯

**Review by cjappl — APPROVED** [review]
Would highly recommend changing the json to be zipped up to save some sanity. :)

File: CMakeLists.txt (hunk 1 above)
Comment (jonmgomez) [reply CMakeLists.txt]: Yup that is much nicer, thanks! Ill follow up with the rest of the tps libraries

File: main.cpp (hunk 2 above)
Comment (jonmgomez) [reply main.cpp:34]: Its only been a month! I have officially lost it 🧠

File: main.cpp [hunk 3]
```diff
@@ -8,12 +8,38 @@
...
 #include <glm/glm.hpp>
+#include <nlohmann/json.hpp>
 
+#include <fstream>
 #include <iostream>
 #include <string>
 
 using namespace pong;
 
+nlohmann::json OpenJsonFile(const std::string& filePath)
```
Comment (cjappl) [inline main.cpp:19]: You'll have to let me know what you think of this library. It is the big json library on the block but I've never used it myself

**Review by cjappl — COMMENTED** [review]
Much better!!

**Pull request merged**